### PR TITLE
IMAP Sensor fixes

### PR DESCRIPTION
**Pull request by jfryman**
This PR fixes two issues with the IMAP sensor.

1) IMAP sensor dies when attempting to read an older message marked as "Unread"
2) IMAP sensor does not properly parse message headers

### CODE REVIEW

File: packs/email/config.yaml [hunk 1]
```diff
@@ -1,16 +1,11 @@
...
-    server: ""
+  steve_main:
+    server: imap.kolabnow.com
     port: 993
-    username: ""
-    password: ""
-    ssl: True
-    download_attachments: True
-
+    username: steve@fryman.io
```
Comment (Kami) [inline packs/email/config.yaml]: I assume you didn't want to include this? :)

File: packs/email/sensors/imap_sensor.py [hunk 2]
```diff
@@ -41,17 +41,18 @@ def __init__(self, sensor_service, config=None, poll_interval=30):
...
         self._logger.debug('[IMAPSensor]: entering setup')
 
-        if 'imap_mailboxes' in self._config:
-            self._parse_mailboxes(self._config['imap_mailboxes'])
-
     def poll(self):
         self._logger.debug('[IMAPSensor]: entering poll')
 
+        if 'imap_mailboxes' in self._config:
+            self._parse_mailboxes(self._config['imap_mailboxes'])
```
Comment (Kami) [inline packs/email/sensors/imap_sensor.py]: Just want to confirm - we re-establish a connection to the IMAP server on every poll now, right?

Comment (jfryman) [inline packs/email/sensors/imap_sensor.py]: Yes.

Comment (jfryman) [inline packs/email/sensors/imap_sensor.py]: Ok! After some :tea:, I am human again.

I put some details on this on commit https://github.com/StackStorm/st2contrib/commit/a5d6e76e6e60616ed5d51b46895cbe89d5bf5a4a. I was having to do deep traces into the Python IMAP library and how it implemented RFCs as I was getting protocol errors. Time to fix vs overhead, I went this way to save some time. If we need to optimize for a long poll, we'll have to ditch the `easyimap` library and get closer to the library and protocol.

**Pull request merged**